src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -502,9 +508,19 @@ private string GenerateForCollection(TypeGenerationSpec typeGenerationSpec)
                         break;
                 }
 
-                string metadataInitSource = @$"_{typeGenerationSpec.TypeInfoPropertyName} = {collectionTypeInfoValue};";
+                string metadataInitSource = @$"{JsonCollectionInfoValuesTypeRef}<{typeRef}> {InfoVarName} = new {JsonCollectionInfoValuesTypeRef}<{typeRef}>()
+                {{
+                    {ObjectCreatorPropName} = {objectCreatorValue},
+                    KeyInfo = {keyTypeMetadataPropertyName!},
```

Comment:
Why is `KeyInfo` and `ElementInfo` not a const like the other properties?

Reply:
I've made them `const` now. Didn't do that initially because they aren't used elsewhere.